Mathlib/Algebra/Module/Zlattice.lean

```diff
@@ -258,3 +303,143 @@ protected theorem isAddFundamentalDomain [Finite ι] [MeasurableSpace E] [OpensM
 end Real
 
 end Zspan
+
+section Zlattice
+
+open Submodule
+
+variable (K : Type*) [NormedLinearOrderedField K] [HasSolidNorm K] [FloorRing K]
+variable {E : Type*} [NormedAddCommGroup E] [NormedSpace K E] [FiniteDimensional K E]
+variable [ProperSpace E] {L : AddSubgroup E} [DiscreteTopology L]
+variable (hs : span K (L : Set E) = ⊤)
+
+theorem Zlattice.FG : AddSubgroup.FG L := by
+  suffices (AddSubgroup.toIntSubmodule L).FG by exact (fg_iff_add_subgroup_fg _).mp this
+  obtain ⟨s, ⟨h_incl, ⟨h_span, h_lind⟩⟩⟩ := exists_linearIndependent K (L : Set E)
+  -- Let `s` be a maximal `K`-linear independent family of elements of `L`. We show that
+  -- `L` is finitely generated (as a ℤ-module) because it fits in the exact sequence
+  -- `0 → span ℤ s → L → L / span ℤ s → 0` with `span ℤ s` and `L / span ℤ s` finitely generated.
+  refine fg_of_fg_map_of_fg_inf_ker (span ℤ s).mkQ ?_ ?_
```

Comment:
I'm inclined to slightly reorganize the proof.
```suggestion
  have spanZsFG : (span ℤ s).FG := by
    rw [ker_mkQ, inf_of_le_right (span_le.mpr h_incl)]
    exact fg_span (LinearIndependent.finite h_lind)
  refine fg_of_fg_map_of_fg_inf_ker (span ℤ s).mkQ ?_ spanZsFG
```
Then you don't need to indent the rest of the proof.

Reply:
Indeed. But the goal to prove is actually a bit more complicated to state than `(span ℤ s).FG` so I prefer to use refine here.